### ISSUE TITLE
Combining MelGAN with Tacotron2 model trained in ESPNET

### ISSUE THREAD

**Issue by ayush9096** [TensorSpeech/TensorFlowTTS#252]
Hi,
I am facing little difficulty in combining the Tacotron2 model trained in Espnet with the MB-MelGAN trained in TensorflowTTS.
I think it is because Espnet applies CMVN for pre-processing, and TensorflowTTS uses Sklearn preprocessing Standard Scalar().
After generating a Mel from Tacotron2 model from espnet , I am applying normalization stored in stats.npy.

After that running inference on MB-MelGAN, the result is not so good in terms of clarity. 
Any Reason for this or I am missing or doing something wrong ?

**Comment by dathudeptrai** [comment]
after generate mel from tacotron2 you need denorm it into raw-mel then apply our normalization.

**Comment by ayush9096** [comment]
I am applying the CMVN after generating mel from ESPNET, and then applying the normalization as in stats.npy but the quality I am getting is not expected.

**Comment by dathudeptrai** [comment]
@ayush9096 i do not know what is CMVN ?, and the preprocessing here is the same as espnet's vocoder used. I think the output of tacotron2 espnet can be use directly with our pretrained.

**Comment by ZDisket** [comment]
@ayush9096 A few months ago I trained a few ESPNet models and used kan-bayashi's ParallelWaveGAN as a vocoder, which can be converted to this repo and used, so vocoders here should be compatible. Maybe you meant ESPNet2? Also, I believe stats normalization is for using with NVIDIA's Tacotron 2.

**Comment by ahmedalbahnasawy** [comment]
Hi @dathudeptrai , @ZDisket , 
My text2mel model trained using Espnet1. My vocoder is MB-Melgan well trained using your's. 
I would like to combine them but the generated audio is weird. 
<img width="751" alt="train" src="https://user-images.githubusercontent.com/50653875/97779977-2a63da80-1b9b-11eb-9f1e-2c8204ec9932.png">
<img width="912" alt="eval" src="https://user-images.githubusercontent.com/50653875/97779988-3cde1400-1b9b-11eb-8991-8a8f87ab9377.png">

```
with torch.no_grad():
    c,_,_ = Espnet-model.inference(k,inference_args) # Espnet text2mel model
c = c.numpy()
c = numpy.expand_dims(c,0) # add 1D 
c = tf.convert_to_tensor(c,dtype=tf.float32)
audio = mb_melgan.inference(c)[0,:,0]
```
What i'm doing wrong ? Thanks

**Comment by ZDisket** [comment]
@ahmedalbahnasawy Were the Tacotron2 and mb-melgan done with the same preprocessing parameters? (same mel_fmin, fmax, etc). Also, can you share audio samples?

**Comment by ahmedalbahnasawy** [comment]
@ZDisket yeah. both of them have the same (fs=16000,fmax=7600, fmin=80, n_mels=80 ,n_fft=1024, n_shift=256 , trimming=True, trim_threshold=60). I think i need to denormalize mels first then normalize it again but i don't know if its correct. 
I have 2 samples here one using text2mel-kan-bayashi vocoder vs text2mel-tts-tf vocoder.
[samples.zip](https://github.com/TensorSpeech/TensorFlowTTS/files/5469714/samples.zip)

**Comment by ahmedalbahnasawy** [comment]
@dathudeptrai Do you think this approach is correct by de-norm my mel-spectrogram based on Espnet stats then re-norm based-on your mb-gan stats. I did the first part by de-norm tacotron mels ( inverse of CMVN) . Then normalize using this repo "stats.npy" .

```
def denorm_mel(mel: np.array) -> np.ndarray:
    cmvn_path = "cmvn.ark"
    cmvn=CMVN(cmvn_path, norm_means=True, norm_vars=True, reverse=True)
    denorm = cmvn(mel)
    return denorm

def norm_tf(mel: np.ndarray) -> np.ndarray:
    scaler = StandardScaler()
    scaler.mean_, scaler.scale_=np.load("stats.npy")
    scaler.n_features_in_ = 80
    mel_norm = scaler.transform(mel)
    return mel_norm
```
The generated samples are lacking clarity and the reading style is not correct.
I have zero doubt about your text2mel models and i'm already training text2mel model from your repo :boom: 
[samples2.zip](https://github.com/TensorSpeech/TensorFlowTTS/files/5494350/samples2.zip)
